tools/check-typo

```diff
@@ -88,13 +70,42 @@
 # First prevent i18n from messing up everything.
 export LC_ALL=C
 
+OCAML_CT_CAT=${OCAML_CT_CAT:-cat}
+OCAML_CT_LS_FILES=${OCAML_CT_LS_FILES:-git ls-files}
+OCAML_CT_HEAD=${OCAML_CT_HEAD:-HEAD}
+OCAML_CT_AWK=${OCAML_CT_AWK:-awk}
+if [ -z "${OCAML_CT_GIT_INDEX+x}" ] ; then
+  OCAML_CT_GIT_INDEX=
+else
+  OCAML_CT_GIT_INDEX="GIT_INDEX_FILE=$OCAML_CT_GIT_INDEX"
+fi
+
+# The output of processing the attributes should be whitespace-separated with
+# - the "typo." prefix dropped
+# - unset/false keys not present
+# - set/true keys present
+# - "may" keys present, suffixed by a question mark
+#
+# for example,
+#   typo.long-line: set
+#   typo.missing-header: may
+#   typo.very-long-line: false
+# should result in "long-line missing-header?"
+get_attrs() {
+  env $OCAML_CT_GIT_INDEX git check-attr --all $OCAML_CT_CA_FLAG "$1" \
+  | grep -o " typo\\..*$" | sed "s/ typo\\.//g" \
+  | grep -v ": unset" | grep -v ": false" \
+  | sed "s/: set//g" | sed "s/: true//g" | sed "s/: may/?/g"
+}
+
 # Special case for recursive call from the find command (see IGNORE_DIRS).
 case "$1" in
   --check-prune)
-    case `env $OCAML_CT_GIT_INDEX git check-attr $OCAML_CT_CA_FLAG \
-                                                 ocaml-typo "$2" 2>/dev/null` in
-      *prune*) echo "INFO: pruned directory $2 (ocaml-typo=prune)" >&2; exit 0;;
-      *) exit 3;;
+    case $2,"$(get_attrs "$2" | grep prune)" in
```

Comment:
Pedantically, do `grep -xF prune`

Reply:
I just check that [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=grep&sektion=&n=1) and [OSX](https://ss64.com/osx/grep.html) both support `-x` and its longer form `--line-regexp`, so I will use that, thanks.

Reply:
(I'm currently confused as to whether `get_attrs` really returns only the value, or may carry some may/must information also on the line.)